routers/repo/issue.go

```diff
@@ -734,6 +734,20 @@ func ViewIssue(ctx *context.Context) {
 		}
 		prConfig := prUnit.PullRequestsConfig()
 
+		if ctx.IsSigned {
+			if err := pull.GetBaseRepo(); err != nil {
+				log.Error(4, "GetBaseRepo: %v", err)
+			} else {
+				// Check whether user may push
+				if protected, err := pull.BaseRepo.IsProtectedBranch(pull.BaseBranch, ctx.User); err != nil {
+					log.Error(4, "IsProtectedBranch: %v", err)
+					ctx.Data["BaseBranchNotProtected"] = false
+				} else {
+					ctx.Data["BaseBranchNotProtected"] = !protected
```

Comment:
Reuse `IsRepositoryWriter && CheckUserAllowedToMerge` to assign to something like `ctx.Data["AllowMerge"]` and check that in `pull.tmpl` with `{{if .AllowMerge}}`